packages/grafana-prometheus/src/components/metrics-browser/MetricsBrowserContext.tsx

```diff
@@ -0,0 +1,370 @@
+import { createContext, PropsWithChildren, useCallback, useContext, useEffect, useMemo, useState } from 'react';
+
+import { TimeRange } from '@grafana/data';
+
+import PromQlLanguageProvider from '../../language_provider';
+
+import { buildSelector } from './selectorBuilder';
+import { DEFAULT_SERIES_LIMIT, EMPTY_SELECTOR, LAST_USED_LABELS_KEY, Metric, METRIC_LABEL } from './types';
+
+/**
+ * Context for the Metrics Browser component
+ * Provides state and handlers for browsing and selecting Prometheus metrics and labels
+ */
+interface MetricsBrowserContextType {
+  // Error and status state
+  err: string;
+  setErr: (err: string) => void;
+  status: string;
+  setStatus: (status: string) => void;
+
+  // Series limit settings
+  seriesLimit: string;
+  setSeriesLimit: (limit: string) => void;
+
+  // Callback when selector changes
+  onChange: (selector: string) => void;
+
+  // Data and selection state
+  metrics: Metric[];
+  labelKeys: string[];
+  labelValues: Record<string, string[]>;
+  selectedMetric: string;
+  selectedLabelKeys: string[];
+  selectedLabelValues: Record<string, string[]>;
+
+  // Event handlers
+  onMetricClick: (name: string) => void;
+  onLabelKeyClick: (name: string) => void;
+  onLabelValueClick: (labelKey: string, labelValue: string) => void;
+  getSelector: () => string;
+  onClearClick: () => void;
+
+  // Validation
+  validationStatus: string;
+  onValidationClick: () => void;
+}
+
+const MetricsBrowserContext = createContext<MetricsBrowserContextType | undefined>(undefined);
+
+type MetricsBrowserProviderProps = {
+  timeRange: TimeRange;
+  languageProvider: PromQlLanguageProvider;
+  onChange: (selector: string) => void;
+};
+
+/**
+ * Filter function to exclude the metric label
+ */
+const withoutMetricLabel = (ml: string) => ml !== METRIC_LABEL;
+
+/**
+ * Provider component for the Metrics Browser context
+ * Manages state and data fetching for metrics, labels, and values
+ */
+export function MetricsBrowserProvider({
+  children,
+  timeRange,
+  languageProvider,
+  onChange,
+}: PropsWithChildren<MetricsBrowserProviderProps>) {
+  // State for UI settings and messages
+  const [seriesLimit, setSeriesLimit] = useState(DEFAULT_SERIES_LIMIT);
+  const [err, setErr] = useState('');
+  const [status, setStatus] = useState('Ready');
```

Comment:
Is this "Ready" here on purpose, cause in the current version, it does not show it. 🤔